### PR TITLE
Improve get user authorized organization service

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
## Purpose
- Improve get user authorized organization service

## Approach
- At the moment, we identify a given organization is an authorized organization if a user association with the authenticated user for that given organization exist. We can improve that logic further to check the associated user has a role associated with him to indicate it is an authorized organization.

## Related Issue
- https://github.com/wso2/product-is/issues/18727

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java [hunk 1]
```diff
@@ -136,17 +136,18 @@ List<BasicOrganization> getOrganizations(Integer limit, String after, String bef
...
+     * @param sortOrder           The sort order, ascending or descending.
+     * @param filter              The filter string.
+     * @param recursive           Determines whether records should be retrieved in a recursive manner.
+     * @param applicationAudience The application audience.
      * @return the list of {@link BasicOrganization}s.
      * @throws OrganizationManagementException The exception thrown when listing organizations.
      */
     List<BasicOrganization> getUserAuthorizedOrganizations(Integer limit, String after, String before, String sortOrder,
-                                                           String filter, boolean recursive)
+                                                           String filter, boolean recursive, String applicationAudience)
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java:150]: let's introduce a new method without changing the exiting method

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java [hunk 2]
```diff
@@ -98,6 +98,15 @@ public class SQLConstants {
...
             "ON UM_ORG.UM_ID = UM_ORG_USER_ASSOCIATION.UM_ORG_ID WHERE ";
 
+    public static final String GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS =
+            "SELECT UM_ORG.UM_ID, UM_ORG.UM_ORG_NAME, UM_ORG.UM_CREATED_TIME, UM_ORG.UM_STATUS FROM " +
+                    "UM_ORG_USER_ASSOCIATION INNER JOIN UM_ORG ON UM_ORG.UM_ID = UM_ORG_USER_ASSOCIATION.UM_ORG_ID " +
+                    "INNER JOIN UM_TENANT ON UM_TENANT.UM_ORG_UUID = UM_ORG.UM_ID " +
+                    "INNER JOIN UM_HYBRID_ROLE ON UM_HYBRID_ROLE.UM_TENANT_ID = UM_TENANT.UM_ID " +
+                    "INNER JOIN UM_HYBRID_USER_ROLE ON UM_HYBRID_USER_ROLE.UM_ROLE_ID = UM_HYBRID_ROLE.UM_ID " +
+                    "INNER JOIN UM_USER ON UM_USER.UM_USER_NAME = UM_HYBRID_USER_ROLE.UM_USER_NAME " +
+                    "INNER JOIN UM_SHARED_ROLE ON UM_HYBRID_ROLE.UM_ID  = UM_SHARED_ROLE.UM_SHARED_ROLE_ID WHERE ";
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java:108]:
```suggestion
                    "INNER JOIN UM_SHARED_ROLE ON UM_HYBRID_ROLE.UM_ID = UM_SHARED_ROLE.UM_SHARED_ROLE_ID WHERE ";
```

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java [hunk 3]
```diff
@@ -699,14 +705,14 @@ public String resolveTenantDomain(String organizationId) throws OrganizationMana
...
-                                                         List<ExpressionNode> parentIdExpressionNodes)
+                                                         List<ExpressionNode> parentIdExpressionNodes,
+                                                         String applicationAudience)
             throws OrganizationManagementServerException {
 
         FilterQueryBuilder filterQueryBuilder = new FilterQueryBuilder();
         appendFilterQuery(expressionNodes, filterQueryBuilder);
         Map<String, String> filterAttributeValue = filterQueryBuilder.getFilterAttributeValue();
         List<String> timestampTypeAttributes = filterQueryBuilder.getTimestampFilterAttributes();
-
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java:709]: let's revert this change

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java (hunk 1 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java:166]: Fixed

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java (hunk 2 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/SQLConstants.java]: Fixed

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java (hunk 3 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java]: Fixed

**Review by dewniMW — APPROVED** [review]
